packages/flutter/lib/src/widgets/scroll_controller.dart

```diff
@@ -166,6 +166,15 @@ class ScrollController extends ChangeNotifier {
     return '$runtimeType#$hashCode(${description.join(", ")})';
   }
 
+  /// Add additional information to the given description for use by [toString].
+  ///
+  /// This method makes it easier for subclasses to coordinate to provide a
+  /// high-quality [toString] implementation. The [toString] implementation on
+  /// the [State] base class calls [debugFillDescription] to collect useful
```

Comment:
we really should factor out these docs into an interface at the foundation layer.